code/modules/mob/living/simple_animal/hostile/viscerator.dm

```diff
@@ -114,3 +114,69 @@
 		var/mob/living/carbon/M = target
 		M.adjustCloneLoss(rand(5,10)) //scp-553 LORE.
 	..()
+
+/mob/living/simple_animal/hostile/bunnybot
+	name = "Bunnybot"
+	desc = "A sloppily assembled wind-up toy with a rabbit face."
+	icon = 'icons/obj/butterfly.dmi'
+	icon_state = "bunnybot"
+	icon_living = "bunnybot"
+	icon_dead = "bunnybot_dead"
+	pass_flags = PASSMOB
+	health = 45
+	maxHealth = 45
+	melee_damage_lower = 0
+	melee_damage_upper = 1 //Aaahhh my toe
+	attack_sound = "sound/items/trayhit2.ogg"
+	speak_emote = list("whirrs")
+	emote_hear = list("whirrs")
+	emote_see = list("whirrs")
+	attacktext = "bumps into"
+	speak_chance = 1
+	turns_per_move = 15
+	response_help  = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm   = "stomps"
+	stop_automated_movement = 1
+	friendly = "bumps into"
+	can_butcher = 0
+	var/autodie = TRUE
+
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	minbodytemp = 0
+
+	size = SIZE_SMALL
+	meat_type = null
+	held_items = list()
+	mob_property_flags = MOB_ROBOTIC
+
+/mob/living/simple_animal/hostile/bunnybot/Life()
+	..()
+	if(autodie && life_tick > 12)
+		death()
+
+/mob/living/simple_animal/hostile/bunnybot/death(var/gibbed = FALSE)
+	playsound(src, "sound/effects/explosion_small1.ogg", 25, 1)
+	icon_state = "bunnybot_dead"
+	animate(src,alpha = 0, time = 3 SECONDS)
```

Comment:
fadeout animation seems like a weird choice for this? why not just do an explosion effect

Reply:
Since it's spammable I didn't want the hallways littered with corpses/wreckage. The viscerator just disappearing didn't work for it since it's bigger so it looks really jarring. It is kind of weird now that you mention it though. I'll change it.

Reply:
What you could do is have the corpse exist for a couple seconds, then after that do the explosion effect and have it disappear immediately.

Reply:
The corpse is meant to be it post-explosion though. Like it's such a poorly made toy it explodes/breaks after a few seconds. I'll change it if you prefer but see how you like the minor change I just made. It looks a lot more natural. The explosion sound makes it seem like it breaks, catches fire, then immediately burns up into nothing.